File: python/diamond/diamond.py
import string
def rows(letter):
    letters = [*string.ascii_uppercase]
    idx = letters.index(letter)
    width = idx * 2 + 1
    left = int(width / 2)
    right = left
    rows = []
    for letter in letters[0:idx + 1]:
        row = ''
        for i in range(width):
            if i == left or i == right:
                row += letter
            else:
                row += ' '
        left -= 1
        right += 1
        rows.append(row)
    rows = rows + rows[-2::-1]
    return rows

File: ruby/diamond/diamond.rb
class Diamond
  def self.pad(length)
    " " * length
  end
  def self.make_diamond(letter)
    letters = [*'A'..letter]
    width = ((letters.size - 1) * 2)
    half = width / 2
    out = letters.map.with_index { |l, i|
      i == 0 ? pad(half) + l + pad(half) : pad(half - i) + l + pad(i*2 - 1) + l + pad(half - i)
    }
    out += out[0..-2].reverse
    out.join("\n") + "\n"
  end
end
